content/en/docs/tasks/tls/manual-rotation-of-ca-certificates.md

```diff
@@ -0,0 +1,137 @@
+
+1. Restart API servers with the certificates (`apiserver.crt`, `apiserver-kubelet-client.crt`, and `front-proxy-client.crt`) signed by new CA. You can use the existing private keys or new private keys. If you changed the private keys then update these in the Kubernetes certificates directory as well.
+
+    Since the pod trusts both old + new CAs, there will be a momentarily disconnection after which the pod's kube client will reconnect to the new API server that uses the certificate signed by the new CA.
+
+    *    Restart Scheduler to use the new CAs.
+
+    *    Make sure control plane components logs no TLS errors.
+
+    {{< note >}}
+
+    To generate certificates and private keys for your cluster using the `openssl` command line tool, see [Certificates (`openssl`)](/docs/concepts/cluster-administration/certificates/#openssl).    
+    You can also use [`cfssl`](/docs/concepts/cluster-administration/certificates/#cfssl).
+
+    {{< /note >}}
+
+1. If your cluster is using bootstrap tokens to join nodes, update the ConfigMap `cluster-info` in the `kube-public` namespace with new CA.
+
+    ```shell
+    base64_encoded_ca="$(base64 /etc/kubernetes/pki/ca.crt)"
+
+    kubectl get cm/cluster-info --namespace kube-public -o yaml | \
+        /bin/sed "s/\(certificate-authority-data:\).*/\1 ${base64_encoded_ca}" | \
+        kubectl apply -f -
+    ```
+
+1. Update the file against `clientCAFile` in kubelet configuration and `certificate-authority-data` in kubelet.conf to use the new CA on all nodes.
+
+    If your kubelet is not using client certificate rotation update `client-certificate-data` and `client-key-data` in kubelet.conf on all nodes along with the kubelet client certificate file usually found in `/var/lib/kubelet/pki`.
+
+1. Follow these steps on all nodes in a rolling fashion.
+
+    1. Restart kubelet to start using the new CA
+
+    1. Restart any other *[aggregated api servers](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/)* or *webhook handlers* to trust the new CA certificates.
+
+    1. Annotate any daemonsets and deployments to trigger new pod creation in a safer rolling fashion.
+
+        Example:
+
+        ```shell
+        for namespace in $(kubectl get namespace -o jsonpath='{.items[*].metadata.name}'); do
+            for name in $(kubectl get deployments -n $namespace -o jsonpath='{.items[*].metadata.name}'); do
+                kubectl patch deployment -n ${namespace} ${name} -p '{"spec":{"template":{"metadata":{"annotations":{"ca-rotation": "1"}}}}}';
+            done
+            for name in $(kubectl get daemonset -n $namespace -o jsonpath='{.items[*].metadata.name}'); do
+                kubectl patch daemonset -n ${namespace} ${name} -p '{"spec":{"template":{"metadata":{"annotations":{"ca-rotation": "1"}}}}}';
+            done
+        done
+        ```
+
+1. Once the cluster functionality is successfully verified:
```

Comment:
This guide doesn't say how cluster functionality is verified. Can you either include a step above saying how someone could verify functionality, or just make this a step about removing the old CA?

Reply:
I guess manually verifying if the control plane and the kubelet logs does not reports tls errors should be fine to understand if the change was accepted. If you have any specific suggestion, i would welcome it.

Reply:
@micahhausler added one step for verification via logs. request you to review again and let me know if I understood and applied all your suggestions correctly.

Reply:
I think this is good enough for now. We can add more comprehensive verification instructions in a later PR